### PR TITLE
Convert math.rsqrt to a function call getRsqrtBf16() for v16bfloat16 and v32bfloat16 types and reorganize files in aie_runtime_lib

### PR DESCRIPTION
**Pull request by linay-xsj**

### CODE REVIEW

File: aie_runtime_lib/AIE2/vec_math.h [hunk 1]
```diff
@@ -0,0 +1,149 @@
+//===-  vec_math.h -====//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2023 Advanced Micro Devices, Inc.
+//
+//
```
Comment (david-vc) [inline aie_runtime_lib/AIE2/vec_math.h:9]: Delete this line.

**Review by david-vc — COMMENTED** [review]
@linay-xsj : Now that you've reorganized aie_runtime_lib, please run all James' TOSA Integration tests to make sure your changes don't break these tests. These tests are NOT run by the build bot.

**Comment by linay-xsj** [comment]
> @linay-xsj : Now that you've reorganized aie_runtime_lib, please run all James' TOSA Integration tests to make sure your changes don't break these tests. These tests are NOT run by the build bot.

James' Integration/Dialect/TOSA tests do not use any of aie_runtime_lib files. Per your request, I tested these TOSA tests. They are all passed except XFAIL ones. I also found some redundant code inside. @jamestcl-amd, could you please clean up the code?

1. test/Integration/Dialect/TOSA/bf16xfloat_add_elem/bf16xfloat_add_elem.mlir
remove line 11-12

2. test/Integration/Dialect/TOSA/i8xi32_mul_elem/i8xi32_mul_elem.mlir
remove --mlir-print-ir-after-all in line 7

**Review by david-vc — APPROVED** [review]
thanks for revising.

**Comment by jamestcl-amd** [comment]
> > @linay-xsj : Now that you've reorganized aie_runtime_lib, please run all James' TOSA Integration tests to make sure your changes don't break these tests. These tests are NOT run by the build bot.
> 
> James' Integration/Dialect/TOSA tests do not use any of aie_runtime_lib files. Per your request, I tested these TOSA tests. They are all passed except XFAIL ones. I also found some redundant code inside. @jamestcl-amd, could you please clean up the code?
> 
> 1. test/Integration/Dialect/TOSA/bf16xfloat_add_elem/bf16xfloat_add_elem.mlir
>    remove line 11-12
> 2. test/Integration/Dialect/TOSA/i8xi32_mul_elem/i8xi32_mul_elem.mlir
>    remove --mlir-print-ir-after-all in line 7

Thanks for checking up, will do. Also, more like a general question, and maybe we can discuss in tomorrow's meeting.. @david-vc @linay-xsj Is there any historical reason that we don't add tests to `test/Targets/AIEVecToCpp` to cover the changes to `aie-translate`? Since the current `dut.cc` inside each test is just a reference, I feed like adding those cpp library call to tests in `test/Targets/AIEVecToCpp` might be helpful?

**Comment by linay-xsj** [comment]
> > > @linay-xsj : Now that you've reorganized aie_runtime_lib, please run all James' TOSA Integration tests to make sure your changes don't break these tests. These tests are NOT run by the build bot.
> > 
> > 
> > James' Integration/Dialect/TOSA tests do not use any of aie_runtime_lib files. Per your request, I tested these TOSA tests. They are all passed except XFAIL ones. I also found some redundant code inside. @jamestcl-amd, could you please clean up the code?
> > 
> > 1. test/Integration/Dialect/TOSA/bf16xfloat_add_elem/bf16xfloat_add_elem.mlir
> >    remove line 11-12
> > 2. test/Integration/Dialect/TOSA/i8xi32_mul_elem/i8xi32_mul_elem.mlir
> >    remove --mlir-print-ir-after-all in line 7
> 
> Thanks for checking up, will do. Also, more like a general question, and maybe we can discuss in tomorrow's meeting.. @david-vc @linay-xsj Is there any historical reason that we don't add tests to `test/Targets/AIEVecToCpp` to cover the changes to `aie-translate`? Since the current `dut.cc` inside each test is just a reference, I feed like adding those cpp library call to tests in `test/Targets/AIEVecToCpp` might be helpful?

You can add tests in test/Targets/AIEVecToCpp for aie-translate, if it is nescessary.

**Comment by jamestcl-amd** [comment]
> > > > @linay-xsj : Now that you've reorganized aie_runtime_lib, please run all James' TOSA Integration tests to make sure your changes don't break these tests. These tests are NOT run by the build bot.
> > > 
> > > 
> > > James' Integration/Dialect/TOSA tests do not use any of aie_runtime_lib files. Per your request, I tested these TOSA tests. They are all passed except XFAIL ones. I also found some redundant code inside. @jamestcl-amd, could you please clean up the code?
> > > 
> > > 1. test/Integration/Dialect/TOSA/bf16xfloat_add_elem/bf16xfloat_add_elem.mlir
> > >    remove line 11-12
> > > 2. test/Integration/Dialect/TOSA/i8xi32_mul_elem/i8xi32_mul_elem.mlir
> > >    remove --mlir-print-ir-after-all in line 7
> > 
> > 
> > Thanks for checking up, will do. Also, more like a general question, and maybe we can discuss in tomorrow's meeting.. @david-vc @linay-xsj Is there any historical reason that we don't add tests to `test/Targets/AIEVecToCpp` to cover the changes to `aie-translate`? Since the current `dut.cc` inside each test is just a reference, I feed like adding those cpp library call to tests in `test/Targets/AIEVecToCpp` might be helpful?
> 
> You can add tests in test/Targets/AIEVecToCpp for aie-translate, if it is nescessary.

I mean for all these cpp library call conversion, perhaps we should have Conversion tests coverage other than the unit tests. The  unit_tests/aievec_tests are very heavy and go through chess and simulator for numerics. As for tests in `Conversion` or `Target`, they are light weight, and can provide instant feedback if changes in `aie-opt` or `aie-translate` break anything.

**Comment by linay-xsj** [comment]
> > > > > @linay-xsj : Now that you've reorganized aie_runtime_lib, please run all James' TOSA Integration tests to make sure your changes don't break these tests. These tests are NOT run by the build bot.
> > > > 
> > > > 
> > > > James' Integration/Dialect/TOSA tests do not use any of aie_runtime_lib files. Per your request, I tested these TOSA tests. They are all passed except XFAIL ones. I also found some redundant code inside. @jamestcl-amd, could you please clean up the code?
> > > > 
> > > > 1. test/Integration/Dialect/TOSA/bf16xfloat_add_elem/bf16xfloat_add_elem.mlir
> > > >    remove line 11-12
> > > > 2. test/Integration/Dialect/TOSA/i8xi32_mul_elem/i8xi32_mul_elem.mlir
> > > >    remove --mlir-print-ir-after-all in line 7
> > > 
> > > 
> > > Thanks for checking up, will do. Also, more like a general question, and maybe we can discuss in tomorrow's meeting.. @david-vc @linay-xsj Is there any historical reason that we don't add tests to `test/Targets/AIEVecToCpp` to cover the changes to `aie-translate`? Since the current `dut.cc` inside each test is just a reference, I feed like adding those cpp library call to tests in `test/Targets/AIEVecToCpp` might be helpful?
> > 
> > 
> > You can add tests in test/Targets/AIEVecToCpp for aie-translate, if it is nescessary.
> 
> I mean for all these cpp library call conversion, perhaps we should have Conversion tests coverage other than the unit tests. The unit_tests/aievec_tests are very heavy and go through chess and simulator for numerics. As for tests in `Conversion` or `Target`, they are light weight, and can provide instant feedback if changes in `aie-opt` or `aie-translate` break anything.

unit_tests/aievec_tests  is to test the correctness of these transformations, while tests in conversion and target are to test whether the transformations are done in the way we wish to.

**Pull request merged**